versioning/versioning.rb

```diff
@@ -7,9 +7,17 @@ class Versioning
   class << self
     def current_version
       verify_git!
-      version=`git describe --tags --abbrev=8 --dirty 2> /dev/null`.strip
-      verify_semver_tag!(version)
-      version.delete_prefix!('v')
+
+      git_describe_version=`git describe --tags --abbrev=8 --dirty 2> /dev/null`.strip
+      unless git_describe_version =~ GIT_SEMVER_REGEX
+        if git_describe_version.include?('+')
+          raise('A git tag version including plus elements is not supported!')
+        else
+          raise('A git tag with an semantic version is required!')
+        end
+      end
```

Comment:
This logic seems wrong (and therefore also shows that the spec tests are not comprehensive, or that I'm confused):

You need to reject all tags that either don't match the regex, or include the `+`. If the tag is not a semver, then we don't care if it includes a `+`, as it is meaningless anyways. But if the tag is a valid semver, then we still need to verify that it doesn't include a `+<build>` element:

```ruby
if git_describe_version =~ GIT_SEMVER_REGEX
  if git_describe_version.include?('+')
    raise('A git tag version including +<build> elements is not supported!')
  end
else
   raise('A git tag with an semantic version is required!')
end
```

Reply:
Sorry, I that it is confusing. I have removed the + part from the SEMVER_REGEX so the logic is correct.
Should I re-add the build part of the semver tag and change the code accordingly or should I rename the regex?